collectors/ovn/metrics.go

```diff
@@ -0,0 +1,168 @@
+package ovn
+
+import (
+	"github.com/openstack-k8s-operators/dataplane-node-exporter/collectors/lib"
+	"github.com/openstack-k8s-operators/dataplane-node-exporter/config"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var openvSwitch = map[string]lib.Metric{
+	"ovn-remote-probe-interval": {
+		Name:        "ovnc_remote_probe_interval",
+		Description: "Maximum number of milliseconds of idle time on connection to the OVN SB DB before sending an inactivity probe message",
+		ValueType:   prometheus.GaugeValue,
+		Set:         config.METRICS_COUNTERS,
+	},
+	"ovn-openflow-probe-interval": {
+		Name:        "ovnc_openflow_probe_interval",
+		Description: "Maximum number of milliseconds of idle time on OpenFlow connection to the OVS bridge before sending an inactivity probe message",
+		ValueType:   prometheus.GaugeValue,
+		Set:         config.METRICS_COUNTERS,
+	},
+}
+
+var openvSwitchBoolean = map[string]lib.Metric{
+	"ovn-monitor-all": {
+		Name:        "ovnc_monitor_all",
+		Description: "Specifies if ovn-controller should monitor all records  of tables in OVN SB DB. The value of 0 means it will conditionally monitor the records that are needed in the current chassis",
+		ValueType:   prometheus.GaugeValue,
+		Set:         config.METRICS_BASE,
+	},
+}
+
+var openvSwitchLabels = map[string]lib.Metric{
+	"ovn-encap-ip": {
+		Name:        "ovnc_encap_ip",
+		Description: "A metric with a constant '1' value labeled by ipadress that specifies the encapsulation ip address configured on that node",
+		Labels:      []string{"encap_ip"},
+		ValueType:   prometheus.GaugeValue,
+		Set:         config.METRICS_BASE,
+	},
+	"ovn-remote": {
+		Name:        "ovnc_sb_connection_method",
+		Description: "A metric with a constant '1' value labeled by sb_connection_method that specifies the ovn-remote value configured on that node",
+		Labels:      []string{"sb_connection_method"},
+		ValueType:   prometheus.GaugeValue,
+		Set:         config.METRICS_BASE,
+	},
+	"ovn-encap-type": {
+		Name:        "ovnc_encap_type",
+		Description: "A metric with a constant '1' value labeled by type that specifies the encapsulation type that a chassis should use to connect to this node",
+		Labels:      []string{"encap_type"},
+		ValueType:   prometheus.GaugeValue,
+		Set:         config.METRICS_BASE,
+	},
+	"ovn-bridge-mappings": {
+		Name:        "ovnc_bridge_mappings",
+		Description: "A metric with a constant '1' value labeled by mapping that specifies a list of key-value pairs that map a physical network name to a local ovs bridge that provides connectivity to that network.",
```

Comment:
Instead of an opaque value:

```
ovnc_bridge_mappings{bridge_mappings="foo:br1,bar:br2"}
```

Wouldn't it be better to have multiple instances of that metric with multiple labels:

```
ovnc_bridge_mapping{network=foo,bridge=br1} 1
ovnc_bridge_mapping{network=bar,bridge=br2} 1
```